pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go

```diff
@@ -201,13 +201,16 @@ func startDistIngestion(
 		return rw.Err()
 	}
 
-	// We now attempt to create initial splits. We currently do
-	// this once during initial planning to avoid re-splitting on
-	// resume since it isn't clear to us at the moment whether
-	// re-splitting is always going to be useful.
-	if !streamProgress.InitialSplitComplete {
+	// We now attempt to issue splits over the topology if we've never done so
+	// (i.e. we're beginning the c2c job), or if we allow splits after a job level
+	// retry.
+	if !streamProgress.InitialSplitComplete || streamingccl.SplitOnRetry.Get(&execCtx.ExecCfg().Settings.SV) {
 		codec := execCtx.ExtendedEvalContext().Codec
```

Comment:
@stevendanna I wonder if we want to add another gate on post initial splits to prevent them from occurring on every retry. We could only trigger them if the previous job level error involved replanning, or if we're resuming after pause, or if the hwm is sufficiently far behind.

Reply:
I like the idea of only resplitting if we also hit a replan since right now the replan is a little bit of a signal that our data distribution changed. It would also prevent us resplitting when we are in an unrelated error loop.